app/Libraries/Markdown/Osu/DocumentProcessor.php

```diff
@@ -6,35 +6,37 @@
 namespace App\Libraries\Markdown\Osu;
 
 use App\Libraries\LocaleMeta;
-use App\Libraries\Markdown\StyleBlock\Element as StyleBlock;
 use App\Libraries\OsuWiki;
-use League\CommonMark\Block\Element as Block;
-use League\CommonMark\EnvironmentInterface;
+use League\CommonMark\Environment\EnvironmentBuilderInterface;
 use League\CommonMark\Event\DocumentParsedEvent;
-use League\CommonMark\Extension\Table as TableExtension;
-use League\CommonMark\Inline\Element as Inline;
+use League\CommonMark\Extension\CommonMark\Node\Block;
+use League\CommonMark\Extension\CommonMark\Node\Inline;
+use League\CommonMark\Node\Block\Paragraph;
+use League\CommonMark\Node\Inline\Text;
+use League\CommonMark\Node\NodeWalkerEvent;
+use League\CommonMark\Node\StringContainerHelper;
+use League\Config\ConfigurationInterface;
 
 class DocumentProcessor
 {
-    public $firstImage;
-    public $title;
-    public $toc;
+    public ?string $firstImage;
+    public ?string $title;
+    public ?array $toc;
 
-    private $environment;
-    private $event;
+    private ConfigurationInterface $config;
+    private ?NodeWalkerEvent $event;
     private $node;
 
-    private $listLevel;
-    private $tocSlugs;
+    private array $tocSlugs;
 
-    private $relativeUrlRoot;
-    private $wikiLocale;
-    private $wikiPathToRoot;
-    private $wikiAbsoluteRootPath;
+    private ?string $relativeUrlRoot;
+    private ?string $wikiLocale;
+    private string $wikiPathToRoot;
+    private string $wikiAbsoluteRootPath;
```

Comment:
this causes error on news page `/home/news/2021-06-02-new-featured-artist-fleshgod-apocalypse` (or comment containing link like `[link label](/wiki/page)`)

though looking again the function probably shouldn't even be run on non-wiki pages in the first place

Reply:
Ah, I missed this.

So `$wikiPathToRoot` and `$wikiAbsoluteRootPath` are nullable?

> though looking again the function probably shouldn't even be run on non-wiki pages in the first place

Which function is this?

Reply:
`fixWikiUrl`

Reply:
Is there any way to check whether it is wiki page or not?

Reply:
Checking the relevant config (the same as the function which sets the variable) should be enough although I noticed it's still kinda needed for news posts

Reply:
Yeah, wiki link also need fixed in news posts.

Reply:
it's fine for the links. It's just the images take one extra redirect when loading the page.

Reply:
Done.